rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala

```diff
@@ -166,6 +170,58 @@ class AppConfigAuth extends Loggable {
 
 }
 
+/*
+ * A trivial extension to the SimpleUrlAuthenticationFailureHandler that correctly log failure
+ */
+class RudderUrlAuthenticationFailureHandler(failureUrl: String) extends SimpleUrlAuthenticationFailureHandler(failureUrl) {
+  override def onAuthenticationFailure(request: HttpServletRequest, response: HttpServletResponse, exception: AuthenticationException): Unit = {
+    LogFailedLogin.warn(exception, request)
+    super.onAuthenticationFailure(request, response, exception)
+  }
+}
+
+object LogFailedLogin {
+
+  def warn(ex: AuthenticationException, request: HttpServletRequest): Unit = {
+    ApplicationLogger.warn(s"Login authentication failed for user '${getUser(ex)}' from IP '${getRemoteAddr(request)}': ${ex.getMessage}")
+  }
+
+  def getUser(ex: AuthenticationException): String = {
+    ex match {
+      case bce:BadCredentialsException =>
+        bce.getAuthentication match {
+          case user: UsernamePasswordAuthenticationToken => user.getName
+          case _                                         => "unknown"
+        }
+      case _ => "unknown"
+    }
+  }
+
+  def getRemoteAddr(request: HttpServletRequest): String = {
+    /*
+     * Of course there is not reliable way to get remote address, because of proxy, forging, etc.
+     */
+
+    // Some interesting header, get from https://gist.github.com/nioe/11477264
+    // the list seems to be roughtly correctly ordered - perhaps we don't want one, but all?
+    val headers = List("X-Forwarded-For", "Proxy-Client-IP", "WL-Proxy-Client-IP", "HTTP_CLIENT_IP", "HTTP_X_FORWARDED_FOR")
+
+    //utility to display correctly the pair header:value
+    def show(t2: Tuple2[String, String]) = t2._1+":"+t2._2
+
+    // always returns the ip addr as seen from the server, and the list of other perhaps interesting info
+    (
+      request.getRemoteAddr ::
```

Comment:
it's a simple string of the IP. Not sure for the format, we have in the message "IP: xxx.xxx.xxx.xxx|header:foo|header:bar". That would feel strange to have "remoteIP" again preprended to that. +